CMakeLists.txt

```diff
@@ -64,9 +84,57 @@ target_compile_definitions(${PROJECT_NAME} PUBLIC -DCFML_FACE_DETECTOR_MODEL_DIR
 set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 17)
 
 if(CLFML_FACE_DETECTOR_BUILD_EXAMPLE_PROJECTS)
-  add_executable(face_roi_demo ${CMAKE_CURRENT_LIST_DIR}/example/face_roi_demo/demo.cpp)
-  target_link_libraries(face_roi_demo PUBLIC face_detector)
+    if (CLFML_ROS2_PACKAGE_BUILD)
+      add_executable(camera_node ${CMAKE_CURRENT_LIST_DIR}/example/ROS_demo/camera_node.cpp)
+      ament_target_dependencies(camera_node
+        rclcpp
+        sensor_msgs
+        cv_bridge
+        std_msgs
+        geometry_msgs
+        std_srvs
+        OpenCV
+      )
+      add_executable(face_detector_node ${CMAKE_CURRENT_LIST_DIR}/example/ROS_demo/face_detector_node.cpp)
+      ament_target_dependencies(face_detector_node
+        rclcpp
+        sensor_msgs
+        cv_bridge
+        std_msgs
+        geometry_msgs
+        std_srvs
+        OpenCV
+      )
+      add_executable(face_detector_viewer ${CMAKE_CURRENT_LIST_DIR}/example/ROS_demo/face_detector_viewer.cpp)
+      ament_target_dependencies(face_detector_viewer
+        rclcpp
+        sensor_msgs
+        cv_bridge
+        std_msgs
+        geometry_msgs
+        std_srvs
+        OpenCV
+      )
+      # older style of target_link_libraries required to comply with ROS2 ament
+      target_link_libraries(camera_node face_detector)
+      target_link_libraries(face_detector_node face_detector)
+      target_link_libraries(face_detector_viewer face_detector)
+    else ()
+      add_executable(face_roi_demo ${CMAKE_CURRENT_LIST_DIR}/example/face_roi_demo/demo.cpp)
+      target_link_libraries(face_roi_demo PUBLIC face_detector)
+    endif()
 endif()
 
+if (CLFML_ROS2_PACKAGE_BUILD)
+  install(TARGETS
+    camera_node
+    face_detector_node
+    face_detector_viewer
+    DESTINATION lib/${PROJECT_NAME})
+endif()
+
+unset(CLFML_FACE_DETECTOR_BUILD_EXAMPLE_PROJECTS CACHE)
 
-unset(CLFML_FACE_DETECTOR_BUILD_EXAMPLE_PROJECTS CACHE)
\ No newline at end of file
+if (CLFML_ROS2_PACKAGE_BUILD)
```

Comment:
Why this extra if? Can it not be added to the previous if?

Reply:
apparently,  ament_package() should be the last call in the CMakeLists, see https://docs.ros.org/en/foxy/How-To-Guides/Ament-CMake-Documentation.html